### PR TITLE
fix : added test samples for fixing lint issues.

### PR DESCRIPTION
**Pull request by imprateeksh**
### Description
The change is to include some dummy code for terraform, a docker file and go lang program to ensure the checks are not skipped in pipeline runs.
Refer [Issue-6845](https://github.ibm.com/goldeneye/issues/issues/6845)

### Release required?

- [x] No release
- [ ] Patch release (`x.x.X`)
- [ ] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### For mergers

- Use a conventional commit message to set the release level. Follow the [guidelines](https://terraform-ibm-modules.github.io/documentation/#/merging.md).
- Include information that users need to know about the PR in the commit message. The commit message becomes part of the GitHub release notes.
- Use the **Squash and merge** option.

### CODE REVIEW

**Comment by imprateeksh** [comment]
Docker and Go has passed. Working on resolving the below skipped/failed steps:
- terraform partial skip messages i.e. `Terraform validate` and `Terraform validate with trivy`
- terragrunt checks

image

**Comment by ocofaigh** [comment]
@imprateeksh Pull the latest git submodule into this Pr and retry: `git submodule update --remote --merge`

**Comment by ocofaigh** [comment]
@imprateeksh Ahh I see whats wrong - we need to update the pipeline to install dependencies - let me push an update now ..

**Comment by ocofaigh** [comment]
We no longer use terragrunt, so I think we can remove the pre commit hook from our yaml file

**Comment by ocofaigh** [comment]
@imprateeksh Terraform validate is skipped because the hook is actually set up to only scan for `files: ^examples/|^patterns/|^solutions/`

I have a GIT issue in our backlog to see if we can find a way to "opt in" to more directories. The reason we don't scan all files is because the hook will not work on a module that does not have a `provider.tf` file in (aka a module). For now, perhaps add the tf code to an `examples` directory?

**Comment by imprateeksh** [comment]
Thanks @ocofaigh It passes now. 

image

**Review by ocofaigh — CHANGES_REQUESTED** [review]
can you move the tf code into a sub directory in examples (maybe `examples/mock-tf-code`)? And then add a README.md with a description saying the code in the directory is used for testing pre-commit hooks.

The following hook is still skipped:
`Add module repository to go.mod......................(no files to check)Skipped`

To address it, can you rename `test` directory to `tests` and add a go.mod file please?

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -39,6 +39,13 @@ jobs:
         run: |
           chown -R $(id -u):$(id -g) $PWD
 
+      # Install dependencies
+      - name: Install dependencies
+        env:
+          TFSWITCH_DIRECTORY: "test/example_tf"
```
[the remark this answers is not in the document]
Comment (imprateeksh) [reply .github/workflows/ci.yml]: sure. Thank you Conall. I have made the changes and modified `TFSWITCH_DIRECTORY`

**Comment by imprateeksh** [comment]
image

**Comment by ocofaigh** [comment]
@imprateeksh The code in the go.mod hook won't be tested unless you add a go.mod file in a directory called `tests` - please add the go code into there

**Comment by imprateeksh** [comment]
> @imprateeksh The code in the go.mod hook won't be tested unless you add a go.mod file in a directory called `tests` - please add the go code into there. Same with go linter - its only currently setup to check golang code in a `tests` directory

I have added `tests` inside `examples` directory. It has passed the checks too. Do you suggest to move the `tests` outside?
image

**Comment by ocofaigh** [comment]
@imprateeksh Are you sure the hooks are working on those files though? Try locally, but looking at the logic in the scripts that the hooks use, I think they only run if files found in `tests` directory. The hook will still be marked as PASS because the script does run but the logic inside the script doesn't do anything. We want to test the script logic here.

**Comment by imprateeksh** [comment]
> @imprateeksh Are you sure the hooks are working on those files though? Try locally, but looking at the logic in the scripts that the hooks use, I think they only run if files found in `tests` directory. The hook will still be marked as PASS because the script does run but the logic inside the script doesn't do anything. We want to test the script logic here.

Confirmed that the script's execution logic is functioning correctly after moving `tests` directory to the root folder.

File: tests/go.mod [hunk 2]
```diff
@@ -0,0 +1,3 @@
+module sample.go
```
Comment (ocofaigh) [inline tests/go.mod:1]: Hmm, if the hook was working, it would actually update this to `module github.ibm.com/GoldenEye/operations`

Comment (imprateeksh) [reply tests/go.mod:1]: @ocofaigh - Please suggest if my understanding is not correct here.

If you are referring to the `terraformDocGoMod.py` file that modifies the `go.mod` file then I think it requires name `module` to be followed by `github` for the above suggested update to occur.

image

**Now, I simply provided `github` after module:** 

image

**This got changed immediately on running pre-commit hook:** 

image

Comment (ocofaigh) [reply tests/go.mod:1]: @imprateeksh ok yes, so can you please add the github url so the code executes

**Review by ocofaigh — APPROVED** [review]

**Pull request merged**